src/material-experimental/mdc-dialog/dialog.scss

```diff
@@ -1,8 +1,34 @@
 @use '@material/dialog' as mdc-dialog;
+@use '@material/dialog/dialog-theme' as mdc-dialog-theme;
 @use '../mdc-helpers/mdc-helpers';
 @use './mdc-dialog-structure-overrides';
 @use '../../cdk/a11y';
 
+// Theme map with values for variables that will be overriden in the theme.
+// MDC's theming system requires non-null values for the slots to be inserted
+// and included as default values.
+$_dialog-initial-theme: (
```

Comment:
How much of this can we extract into a common location shared between multiple components? E.g. `container-color` and `supporting-text-color` seem like they'd be the common surface and on-surface colors for a lot of components.